packages/astro/src/assets/fonts/vite-plugin-fonts.ts

```diff
@@ -0,0 +1,237 @@
+import type { Plugin } from 'vite';
+import type { AstroSettings } from '../../types/astro.js';
+import { resolveProviders } from './providers.js';
+import * as unifont from 'unifont';
+import type { FontFamily, FontProvider } from './types.js';
+import xxhash from 'xxhash-wasm';
+import { extname } from 'node:path';
+import { getBuildOutputDir } from '../../core/build/util.js';
+import { mkdirSync, writeFileSync } from 'node:fs';
+import { generateFontFace } from './utils.js';
+import {
+	DEFAULTS,
+	VIRTUAL_MODULE_ID,
+	RESOLVED_VIRTUAL_MODULE_ID,
+	URL_PREFIX,
+	CACHE_DIR,
+} from './constants.js';
+import { removeTrailingForwardSlash } from '@astrojs/internal-helpers/path';
+import type { Logger } from '../../core/logger/core.js';
+import { createCache, createStorage, type Cache } from './cache.js';
+
+interface Options {
+	settings: AstroSettings;
+	sync: boolean;
+	logger: Logger;
+}
+
+type PreloadData = Array<{ url: string; type: string }>;
+
+const createLogManager = (logger: Logger) => {
```

Comment:
You might want to add some comment that at least explains what's this function for, because it's not intuitive to have "Log manager" that exposes "add" and "remove"